### PR TITLE
Upgrades to pandoc-types 1.22

### LINKED ISSUE #2: Supporting pandoc-types 1.22
Just trying to putting some potentially useful info here. No pressure:

# Links on new table AST

- PR
    - Feature request started here: jgm/pandoc#1024, jgm/pandoc-types#65
    - this is where the actual change of the AST happened: https://github.com/jgm/pandoc-types/pull/66/files?file-filters%5B%5D=.hs#diff-01f4ffe52cf097ab9ff89eebce394c86
    - pandoc 2.10 release note: https://github.com/jgm/pandoc/releases/tag/2.10
    - pandoc 2.11 release note (see HTML writer): https://github.com/jgm/pandoc/releases/tag/2.11
    - pandoc-types changelog: https://github.com/jgm/pandoc-types/blob/master/changelog
- def
    - https://hackage.haskell.org/package/pandoc-types-1.22/docs/Text-Pandoc-Definition.html
    - source: https://github.com/jgm/pandoc-types/blob/master/src/Text/Pandoc/Arbitrary.hs
    - doc on lua filters has some info on this: https://pandoc.org/lua-filters.html
    - See comments at https://github.com/jgm/pandoc-types/pull/66#issuecomment-611053332 and after on normalizing tables
- example ASTs:
    - https://github.com/jgm/pandoc-types/blob/master/test/test-pandoc-types.hs
    - https://github.com/jgm/pandoc/tree/master/test/tables

### PR DESCRIPTION
**Pull request by nokome**
This closes #2 by adding the `Inline::Underline` variant and several changes to the `Block::Table` variant. I think most of those changes are straight forward. Perhaps the trickiest new type is `enum ColWidth` for which I added a new test case.

This also migrates CI to GitHub Workflows (given that "building on travis.org is ceased") and runs tests against Pandoc 2.14.03.

### CODE REVIEW

**Comment by elliottslaughter** [comment]
Thanks for this, will try to look at it soon.

File: README.md [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# Rust port of pandoc-types [![Build Status](https://travis-ci.org/elliottslaughter/rust-pandoc-types.svg?branch=master)](https://travis-ci.org/elliottslaughter/rust-pandoc-types)
+# Rust port of pandoc-types ![Build status](https://github.com/nokome/rust-pandoc-types/actions/workflows/test.yml/badge.svg)
```
Comment (elliottslaughter) [inline README.md:1]: This would need to change to point to the original repo.

**Review by elliottslaughter — APPROVED** [review]
Overall looks good! Just one note to change in the badge URL.

Thanks again for submitting this!

File: README.md (hunk 1 above)
Comment (nokome) [reply README.md]: Fixed.

**Comment by elliottslaughter** [comment]
Merged, thanks again!

**Pull request merged**